fred2/initialstatus.cpp

```diff
@@ -459,15 +479,17 @@ void initial_status::OnOK()
 		MODIFY(Objects[cur_object_index].shield_quadrant[0], (float) m_shields);
 		MODIFY(Objects[cur_object_index].hull_strength, (float) m_hull);
 
-		Objects[cur_object_index].flags.set(Object::Object_Flags::No_shields, m_has_shields != 0);
-
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Force_shields_on, m_force_shields == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Ship_locked, m_ship_locked == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Weapons_locked, m_weapons_locked == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Primaries_locked, m_primaries_locked == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Secondaries_locked, m_secondaries_locked == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Lock_all_turrets_initially, m_turrets_locked == 1);
-        Ships[m_ship].flags.set(Ship::Ship_Flags::Afterburner_locked, m_afterburner_locked == 1);
+		Objects[cur_object_index].flags.set(Object::Object_Flags::No_shields, m_has_shields == 0);
```

Comment:
It's important because `value` is not a real boolean value since it can also have the value 2 for when the editor detected an inconsistent flag state when multiple objects are selected. In that case `handle_inconsistent_flag` does nothing which is the same behavior as before the flagset change.